### PR TITLE
BaSP-TG: tasks methods (create and edit)

### PR DESCRIPTION
**Pull request by DinoStampella**
1) Code the logic to create tasks
2) Code the logic to edit tasks

### CODE REVIEW

File: src/resources/tasks.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const express = require('express');
+const fs = require('fs');
+const tasks = require('../data/tasks.json');
+
+const router = express.Router();
+
+// CREATE TASK
+router.post('/add', (req, res) => {
+  const newTask = req.body;
+  if (Object.keys(newTask).length !== 2) {
```
Comment (Licamus) [inline src/resources/tasks.js:10]: What happens if you enter a wrong key? The task is storages anyway?

Comment (DinoStampella) [reply src/resources/tasks.js:10]: Si Maxi! Creo que no era necesario hacer validaciones para esta problemática

File: src/index.js [hunk 2]
```diff
@@ -13,6 +13,12 @@ app.get('/', async (req, res) => {
   res.send('Hello World!');
 });
 
+// Dino Stampella Workspace
```
Comment (baudraccoverona) [inline src/index.js:16]: Please remove this comment

File: src/index.js [hunk 3]
```diff
@@ -13,6 +13,12 @@ app.get('/', async (req, res) => {
...
+const tasksRouter = require('./resources/tasks');
+
+app.use('/tasks', tasksRouter);
+// End workspace (DS)
```
Comment (baudraccoverona) [inline src/index.js:20]: Remove this one too 😄 🧹

File: src/resources/tasks.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    return;
+  }
+  const newID = tasks[tasks.length - 1].id + 1;
+  newTask.id = newID;
+  tasks.push(newTask);
+  fs.writeFile('src/data/tasks.json', JSON.stringify(tasks), (err) => {
+    if (err) {
+      res.send('Can not save new task');
+    } else {
+      res.send('Task created successfully');
```
Comment (baudraccoverona) [inline src/resources/tasks.js:21]: We should improve this error message to be something like this "Task created successfully. ${newTask}"

File: src/resources/tasks.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+});
+
+// EDIT TASKS
+router.put('/edit/:id', (req, res) => {
+  const taskId = req.params.id;
+  if (Object.keys(req.body).length !== 2) {
+    res.send('Please, complete all fields');
+    return;
+  }
+  const oldTask = tasks.find(((task) => task.id === Number(taskId)));
```
Comment (baudraccoverona) [inline src/resources/tasks.js:33]: We're not handling here the possibility that the task the user wants to edit doesn't exists so we should ask if `oldTasks` exists and if it doesn't we should show an error message like "Task with the id ${taskId} not found"

File: src/resources/tasks.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  const index = tasks.indexOf(oldTask);
+  tasks[index] = req.body;
+  tasks[index].id = Number(taskId);
+  fs.writeFile('src/data/tasks.json', JSON.stringify(tasks), (err) => {
+    if (err) {
+      res.send('Can not edit task');
```
Comment (baudraccoverona) [inline src/resources/tasks.js:39]: We should improve this error message to be something like "Could not edit task with the id ${taskId}"

File: src/resources/tasks.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    } else {
+      res.send('Task edited successfully');
```
Comment (baudraccoverona) [inline src/resources/tasks.js:41]: We should improve this success message to be something like this "Task edited successfully ${task[index].id}"

**Review by baudraccoverona — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (DinoStampella) [reply src/index.js:16]: Comment removed

File: src/index.js (hunk 3 above)
Comment (DinoStampella) [reply src/index.js:20]: Comment removed

File: src/resources/tasks.js (hunk 5 above)
Comment (DinoStampella) [reply src/resources/tasks.js:33]: Validation created

File: src/resources/tasks.js (hunk 6 above)
Comment (DinoStampella) [reply src/resources/tasks.js:39]: Msg updated

File: src/resources/tasks.js (hunk 7 above)
Comment (DinoStampella) [reply src/resources/tasks.js:41]: Msg updated

File: src/resources/tasks.js (hunk 4 above)
Comment (DinoStampella) [reply src/resources/tasks.js:21]: Msg updated

**Review by baudraccoverona — APPROVED** [review]
Looking good! 🚀

**Pull request merged**